src/GregClient/PackageManagerClient.cs

```diff
@@ -0,0 +1,150 @@
+﻿using RestSharp;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Greg.Requests;
+using Greg.Responses;
+using System.Net;
+using System.IO;
+using Greg.Utility;
+using Newtonsoft.Json;
+
+namespace Greg
+{
+    /// <summary>
+    /// Package Manager REST Client for ACG API
+    /// </summary>
+    public class PackageManagerClient : IGregClient
+    {
+        private const string AFC_CODE = "DY1ON1";
+        //Client for ACG REST API
+        private readonly RestClient _client;
+        //Client for file upload, the server address for file upload is different than ACG API
+        private readonly RestClient _fileClient;
+
+        public string BaseUrl { get { return _client.BaseUrl.ToString(); } }
+
+        public readonly IAuthProvider _authProvider;
+        public IAuthProvider AuthProvider
+        {
+            get { return _authProvider; }
+        }
+
+        public PackageManagerClient(IAuthProvider provider, string packageManagerUrl, string fileStorageUrl)
+        {
+            _authProvider = provider;
+            _client = new RestClient(packageManagerUrl);
+            _fileClient = new RestClient(fileStorageUrl);
+        }
+
+        /// <summary>
+        /// Execute the REST request
+        /// </summary>
+        /// <param name="m"></param>
+        /// <returns></returns>
+        private IRestResponse ExecuteInternal(Request m)
```

Comment:
Since this is a private method, you can accept PackageManagerRequest and do the casting while calling this method. Inside this method do the null check.